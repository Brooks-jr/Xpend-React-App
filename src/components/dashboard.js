import React from 'react';
import ExpenseList from './expense-list';
import ExpenseListFilters from './expense-list-filters';

const DashboardPage = () => (
    <div>
        <ExpenseListFilters />
        <ExpenseList />
    </div>
);

export default DashboardPage;