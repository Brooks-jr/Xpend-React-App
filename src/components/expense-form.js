import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';

const now = moment();
// console.log(now.format('Do MMM, YYYY'));

export default class ExpenseForm extends React.Component {
    // default states
    constructor(props) {
        super(props);
        this.state = {
            description: props.expense ? props.expense.description : '',
            note: props.expense ? props.expense.note : '',
            amount: props.expense ? (props.expense.amount / 100).toString() : '',
            createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
            calendarFocused: false,
            error: ''
        };
    };


    // DESCRIPTION on change HANDLER
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };

    // NOTE on change HANDLER
    onNoteChange = (e) => {
        const note = e.target.value;
        this.setState(() => ({ note }));
    };
    // AMOUNT on change HANDLER
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    };

    // DATE on change HANDLER
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }));
        }
    };

    // FOCUSED on change HANDLER
    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    };

    // SUBMIT HANDLER
    onSubmit = (e) => {
        e.preventDefault();
        // ERROR
        if (!this.state.description || !this.state.amount) {
            this.setState(() => ({ error: '*Description and Amount Fields Required' }));
        } else {
            this.setState(() => ({ error: '' }));
            this.props.onSubmit({
                description: this.state.description,
                amount: parseFloat(this.state.amount, 10) * 100,
                createdAt: this.state.createdAt.valueOf(),
                note: this.state.note
            });
        }
    };

    render() {
        return (
            <div>
                {this.state.error && <p>{this.state.error}</p>}
                <form onSubmit={this.onSubmit}>
                    <input type="text" placeholder="Description" autoFocus value={this.state.description} onChange={this.onDescriptionChange} />
                    <input type="text" placeholder="Amount" value={this.state.amount} onChange={this.onAmountChange} />
                    <SingleDatePicker date={this.state.createdAt} onDateChange={this.onDateChange} focused={this.state.calendarFocused} onFocusChange={this.onFocusChange} numberOfMonths={1} isOutsideRange={() => false} />
                    <textarea cols="40" rows="10" placeholder="Note (optional)" value={this.state.note} onChange={this.onNoteChange}></textarea>
                    <button>Add+</button>
                </form>
            </div>
        )
    }
}
