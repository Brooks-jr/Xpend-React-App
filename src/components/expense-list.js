import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './expense-list-item';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) => (
    <div className="container">
        <div className="list-header">
            <div>Expense</div>
            <div>Amount</div>
        </div>

        <div className="list-wrap">
            {
                props.expenses.length === 0 ? (
                    <div className="list-item list--message">
                        <h3>No Expenses Found</h3>
                    </div>

                ) : (
                        props.expenses.map((expense) => {
                            return <ExpenseListItem key={expense.id} {...expense} />
                        })
                    )
            }
        </div>
    </div>
);

const mapStateToProps = (state) => {
    return {
        expenses: selectExpenses(state.expenses, state.filters)
    };
};

export default connect(mapStateToProps)(ExpenseList);
