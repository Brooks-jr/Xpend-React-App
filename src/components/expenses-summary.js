import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
    const formattedExpensesTotal = numeral(expensesTotal / 100).format('$0,0.00');

    return (
        <div className="page-header">
            <div className="container">
                <h1 className="page-header__text">Displaying <span>{expenseCount}</span> {expenseWord} with a sum of <span>{formattedExpensesTotal}</span></h1>
                <div className="page-header__link">
                    <Link className="button button__hover" to="/create">Add Expense</Link>
                </div>
            </div>
        </div>
    );
};

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: selectExpensesTotal(visibleExpenses)
    }
};

export default connect(mapStateToProps)(ExpensesSummary);