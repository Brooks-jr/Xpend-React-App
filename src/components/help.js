import React from 'react';

const HelpPage = () => (
    <div>Help component page!</div>
);

export default HelpPage;