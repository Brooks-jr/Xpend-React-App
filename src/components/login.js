import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
    <div className="box-layout">
        <div className="box-layout__inner-box">
            <h1 className="inner-box__header"><span>X</span>pend</h1>
            <p className="inner-box__text">Start managing your expenses by signing in!</p>
            <button className="button inner-box__button" onClick={startLogin}>Sign In With <span>Google</span></button>
            <p className="inner-box__github">Xpend @ github.com/Brooks-jr/Xpend-React-App</p>
        </div>
    </div>
);

const mapDispatchToProps = (dispatch) => ({
    startLogin: () => dispatch(startLogin())
});

export default connect(undefined, mapDispatchToProps)(LoginPage);