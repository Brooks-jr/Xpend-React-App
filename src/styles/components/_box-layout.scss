.box-layout {
    color: $white;
    display: flex;
    align-items: center;
    background: url('/images/bg-login.jpg');
    background-size: cover;
    height: 100vh;
    width: 100vw;
    justify-content: center;
}

.box-layout__inner-box {
    background: fade-out( $black, .3);
    border-radius: 3px;
    text-align: center;
    padding: $large;
    width: 25rem;
}

.inner-box__header {
    font-size: $xlarge;
    font-weight: 300;
    margin: 0 0 $large 0;
    line-height: 1;
    letter-spacing: 3px;
    span {
        background: -webkit-linear-gradient($logo-blue, $primary);
        background-clip: text;
        -webkit-background-clip: text;
        -webkit-text-fill-color: transparent;
    }
}

.inner-box__text {
}

.inner-box__button {
    &:hover {
        span {
        color: $google-blue;
        }
    }
}

.inner-box__github {
    font-size: $small;
    color: rgba(255, 255, 255, .25);
}