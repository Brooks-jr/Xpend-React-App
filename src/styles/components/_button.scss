.button {
    display: inline-block;
    margin: $medium 0;
    padding: $medium;
    font-size: 2rem;
    font-weight: 500;
    line-height: 1;
    color: $white;
    background: -webkit-linear-gradient(rgb(255, 68, 130), $primary);
    border: none;
    text-decoration: none;
}

.button__hover {
    &:hover {
        background: $white;
        color: $secondary;
    }
}

.button--link {
    background: transparent;
    &:hover {
        color: $primary;
    }
}