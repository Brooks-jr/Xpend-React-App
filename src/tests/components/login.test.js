import React from 'react';
import { shallow } from 'enzyme';
import { LoginPage } from '../../components/login';

test('should render login page', () => {
    const wrapper = shallow(<LoginPage />);
    expect(wrapper).toMatchSnapshot();
});

test('should call login action on click', () => {
    const startLogin = jest.fn();
    const wrapper = shallow(<LoginPage startLogin={startLogin}/>);
    wrapper.find('button').simulate('click');
    expect(startLogin).toHaveBeenCalled();
});