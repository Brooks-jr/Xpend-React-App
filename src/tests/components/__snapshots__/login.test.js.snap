// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render login page 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__inner-box"
  >
    <h1
      className="inner-box__header"
    >
      <span>
        X
      </span>
      pend
    </h1>
    <p
      className="inner-box__text"
    >
      Start managing your expenses by signing in!
    </p>
    <button
      className="button button__hover inner-box__button"
    >
      Sign In With 
      <span>
        Google
      </span>
    </button>
    <p
      className="inner-box__github"
    >
      Xpend @ github.com/Brooks-jr/Xpend-React-App
    </p>
  </div>
</div>
`;
